### PR TITLE
Crankier: Connect to Azure SignalR

### PR DESCRIPTION
**Pull request by staff0rd**
Run Crankier Server using Azure SignalR with;
```
dotnet run -- server --azure-signalr-connectionstring <connection-string>
```

Also, any future for Crankier considering [this](https://github.com/Azure/azure-signalr-bench/)?

### CODE REVIEW

**Comment by dougbu** [comment]
@BrennanConroy @halter73 @mikaelm12 I defer to you on SignalR changes.

@Pilchie this introduces a new Microsoft.Azure.SignalR dependency in this repo. I don't think we use https://github.com/SignalR/SignalR/ packages elsewhere. Am I correct the new dependency is fine in our 'master' branch?

**Comment by Pilchie** [comment]
I don't know of a reason why this would be a problem in master branch.

File: src/SignalR/perf/benchmarkapps/Crankier/Commands/ServerCommand.cs [hunk 1]
```diff
@@ -30,18 +32,28 @@ public static void Register(CommandLineApplication app)
...
         {
             Console.WriteLine($"Process ID: {Process.GetCurrentProcess().Id}");
 
-            var config = new ConfigurationBuilder()
-                .AddEnvironmentVariables(prefix: "ASPNETCORE_")
-                .Build();
+            var configBuilder = new ConfigurationBuilder()
+                .AddEnvironmentVariables(prefix: "ASPNETCORE_");
+
+            if (azureSignalRConnectionString != null)
```
Comment (halter73) [inline src/SignalR/perf/benchmarkapps/Crankier/Commands/ServerCommand.cs]: nit: put even single-line if bodies in brackets

File: src/SignalR/perf/benchmarkapps/Crankier/Server/Startup.cs [hunk 2]
```diff
@@ -11,15 +12,23 @@ namespace Microsoft.AspNetCore.SignalR.Crankier.Server
     public class Startup
     {
         private readonly IConfiguration _config;
+        private readonly string _azureSignalrConnectionString;
         public Startup(IConfiguration configuration)
         {
             _config = configuration;
+            _azureSignalrConnectionString = configuration.GetSection("Azure:SignalR").GetValue<string>("ConnectionString", null);
+            if (_azureSignalrConnectionString != null)
+                Console.WriteLine("Using Azure SignalR");
```
Comment (halter73) [inline src/SignalR/perf/benchmarkapps/Crankier/Server/Startup.cs]: nit: if brackets

It's probably better to put this WriteLine in ServerCommand.Execute anyway.

File: src/SignalR/perf/benchmarkapps/Crankier/Server/Startup.cs [hunk 3]
```diff
@@ -11,15 +12,23 @@ namespace Microsoft.AspNetCore.SignalR.Crankier.Server
...
         }
 
         public void ConfigureServices(IServiceCollection services)
         {
-            var signalrBuilder = services.AddSignalR()
-                .AddMessagePackProtocol();
+            var signalrBuilder = services.AddSignalR();
+
+            if (_azureSignalrConnectionString != null)
+                signalrBuilder.AddAzureSignalR();
```
Comment (halter73) [inline src/SignalR/perf/benchmarkapps/Crankier/Server/Startup.cs]: nit: if brackets

File: src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj [hunk 4]
```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>$(DefaultNetCoreTargetFramework)</TargetFramework>
     <RootNamespace>Microsoft.AspNetCore.SignalR.CranksRevenge</RootNamespace>
+    <DisableTransitiveFrameworkReferences>true</DisableTransitiveFrameworkReferences>
```
Comment (halter73) [inline src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj]: Why is this necessary? It doesn't look like it's something that's used often.

Comment (staff0rd) [reply src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj]: [Version conflict](https://github.com/Azure/azure-signalr/issues/641) with Azure SignalR SDK

**Comment by jkotalik** [comment]
@staff0rd now that the Azure SignalR Service SDK 1.1.x has shipped, you should be able to update this PR. 

@halter73 to follow up.

**Comment by anurse** [comment]
In order to merge we still need:

* Rebasing
* Update to latest Azure SignalR SDK.

Then I think we can take this change!

**Comment by anurse** [comment]
ping @halter73 @BrennanConroy (since I just learned that editing comments to add mentions doesn't actually ping people 🤯)

**Comment by staff0rd** [comment]
ah - i forgot about this, will finish it off this weekend

**Comment by anurse** [comment]
@staff0rd no worries, just checking in :). Worst case, if it goes stale for a while (a few weeks) we may close it to keep our triage queue clear but would be quite happy for you to reopen it when you're ready.

**Comment by staff0rd** [comment]
rebased & updated

File: src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj (hunk 4 above)
Comment (halter73) [reply src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj]: Do you know if this is still necessary?

Comment (halter73) [reply src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj]: Do you know if this is still necessary?

Comment (staff0rd) [reply src/SignalR/perf/benchmarkapps/Crankier/Crankier.csproj]: Seems to still not build without it :/

**Review by halter73 — APPROVED** [review]

**Comment by jkotalik** [comment]
@halter73 will verify to make sure this works _one_ more time.

**Comment by anurse** [comment]
@halter73 did you verify this?

**Comment by halter73** [comment]
Not yet. I'll try to get to this today.

**Comment by halter73** [comment]
Thanks @staff0rd!
